corehq/apps/es/queries.py

```diff
@@ -0,0 +1,38 @@
+import re
+
+
+def match_all():
+    """
+    No-op query used because a default must be specified
+    """
+    return {"match_all": {}}
+
+
+def _smart_query_string(search_string):
+    special_chars = ['&&', '||', '!', '(', ')', '{', '}', '[', ']', '^', '"',
```

Comment:
Sounds good, do you know what that would look like?  A couple of these are two-character things, so I can't just use a character class, and they're all kind of obnoxious special characters that likely need escaping.